File: backend/Sapiforge.Service/CollectionService.cs
namespace Sapiforge.Service;

using Sapiforge.Domain.Interfaces;
using Sapiforge.Domain.Models;

/// <summary>
/// Koleksiyon yönetimi iş mantığını yönetir.
/// ICollectionService interface'ini implemente eder.
/// </summary>
public class CollectionService : ICollectionService
{
    private readonly ICollectionRepository _collectionRepository;

    /// <summary>Bağımlılık dependency injection ile enjekte edilir</summary>
    public CollectionService(ICollectionRepository collectionRepository)
    {
        _collectionRepository = collectionRepository;
    }

    /// <summary>Tüm koleksiyonları getirir</summary>
    public async Task<IEnumerable<Collection>> GetAllAsync()
    {
        return await _collectionRepository.GetAllAsync();
    }

    /// <summary>ID'ye göre koleksiyonu içindeki isteklerle birlikte getirir</summary>
    public async Task<Collection?> GetByIdAsync(int id)
    {
        return await _collectionRepository.GetByIdWithItemsAsync(id);
    }

    /// <summary>Yeni koleksiyon oluşturur</summary>
    public async Task<Collection> CreateAsync(Collection collection)
    {
        return await _collectionRepository.AddAsync(collection);
    }

    /// <summary>
    /// Koleksiyona yeni istek ekler.
    /// Sıralama otomatik olarak sonraki değere atanır.
    /// </summary>
    public async Task AddItemAsync(int collectionId, int requestId)
    {
        var item = new CollectionItem
        {
            CollectionId = collectionId,
            ApiRequestId = requestId
        };
        await _collectionRepository.AddItemAsync(item);
    }

    /// <summary>Koleksiyonu ve içindeki tüm öğeleri siler</summary>
    public async Task DeleteAsync(int id)
    {
        await _collectionRepository.DeleteAsync(id);
    }
}

File: frontend/sapiforge-client/src/services/collectionService.js
import api from './api';

// ── Koleksiyon servisi ──────────────────────────────────────────
// Backend CollectionController ile iletişim kurar

// Tüm koleksiyonları getirir
export const getAllCollections = async () => {
  const response = await api.get('/Collection');
  return response.data;
};

// ID'ye göre koleksiyonu getirir
export const getCollectionById = async (id) => {
  const response = await api.get(`/Collection/${id}`);
  return response.data;
};

// Yeni koleksiyon oluşturur
export const createCollection = async (collectionData) => {
  const response = await api.post('/Collection', collectionData);
  return response.data;
};

// Koleksiyona istek ekler
export const addItemToCollection = async (collectionId, requestId) => {
  await api.post(`/Collection/${collectionId}/items`, requestId);
};

// Koleksiyonu siler
export const deleteCollection = async (id) => {
  await api.delete(`/Collection/${id}`);
};